account_payment_order/models/bank_payment_line.py

```diff
@@ -58,6 +58,17 @@ class BankPaymentLine(models.Model):
         'res.currency',
         related='order_id.payment_mode_id.company_id.currency_id',
         readonly=True, store=True)
+    purpose_code = fields.Selection(selection=[
+        ("SALA", "Salary payment"),
+        ("PENS", "Pension payment"),
+        ("CBFF", "Capital-forming payment (VWL)"),
+        ("CBFR", "Capital-forming payment for retirement (AVWL)"),
+        ("SSBE", "Social Security Benefit"),
+        ("INTC", "ICC payment"),
+        ("CASH", "Cash Management transfer"),
+        ("SUPP", "Other payment"),
+        ("NETT", "Netting"),
+    ], string="Purpose Code")
```

Comment:
Don't you need to add this purpose code on `account.payment.line`, and use it as [grouping criteria](https://github.com/OCA/bank-payment/blob/9d0c6e1c3d673b8eb8040162158fef7dd561083b/account_payment_order/models/bank_payment_line.py#L62) to generate the `bank.payment.line`? Indeed, users don't normally generate bank payment lines manually.